common/dynamicconfig/constants.go

```diff
@@ -570,7 +570,12 @@ is currently processing a task.
 	)
 
 	// keys for frontend
-
+	FrontendHTTPAllowedHosts = NewGlobalTypedSetting(
+		"frontend.httpAllowedHosts",
+		[]string{"*"},
```

Comment:
you can't use a non-nil reference for the default value because of how the default converter works (it does a shallow copy and then asks mapstructure to fill it in). this has to be `[]string(nil)` and then handle the nil in application code (or in your second level converter)

Reply:
I know :(  It should be commented somewhere but I'm not sure where, there's no obvious place for the "api" for defining settings.

I was planning to add a linter or a runtime check using reflection at static init time.